src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -133,48 +162,53 @@ object LDPrune {
     }
 
     if (packIndex < nPacks)
-      a(packIndex) = pack
+      rvb.addLong(pack)
+
+    rvb.endArray()
 
     val nPresent = nSamples - nMissing
     val allHomRef = gtSum == 0
     val allHet = gtSum == nPresent && gtSumSq == nPresent
     val allHomVar = gtSum == 2 * nPresent
 
-    if (allHomRef || allHet || allHomVar || nMissing == nSamples)
-      None
-    else {
+    if (allHomRef || allHet || allHomVar || nMissing == nSamples) {
+      rvb.addInt(nSamples)
+      rvb.addDouble(0d) // placeholder values
+      rvb.addDouble(0d) // placeholder values
+      false
```

Comment:
You can end building a region value early by calling `RegionValueBuilder.clear()`.